### DISCUSSION TITLE
What is the difference between PageModule and Module entities?

### DISCUSSION THREAD

**Discussion by vnetonline** [oqtane/oqtane.framework#3061] [Q&A] [answered]
What is the difference between PageModule and Module entities?

**Comment by leigh-pointer** [comment]
Module ent is the module and a pagemodule is the reference to a module on a page!  PageModule will construct the page with modules

**Comment by sbwalker** [comment]
A module instance in a site has a distinct ModuleId. A module instance can be displayed on multiple pages within a site, or even multiple times on the same page. Each occurrence will have a unique PageModuleId.

**Comment by vnetonline** [comment]
So when you are using ContainerSetting you should use ModuleId if you have a setting Container Specific for a module type?

like this 

`settings = SettingService.SetSetting(settings, GetType().Namespace + ":LinkName:" + ModuleState.ModuleId, _linkName);`
`settings = SettingService.SetSetting(settings, GetType().Namespace + ":LinkTo:" + ModuleState.ModuleId, _linkTo);`

or should you use **PageModuleId**

**Reply by leigh-pointer** [reply, marked as answer]
@vnetonline  the way you currently save is correct.
